packages/medusa/src/services/customer.js

```diff
@@ -385,6 +385,12 @@ class CustomerService extends BaseService {
         customer.metadata = this.setMetadata_(customer, metadata)
       }
 
+      if (email && customer.has_account) {
+        throw new MedusaError(
+          MedusaError.Types.INVALID_DATA,
+          "Email cannot be changed when the user has registered their account"
+        )
+      }
```

Comment:
I don't think this should be enforced at the service level. Let me try to explain:

If a customer has registered it seems intrusive that an admin can change the customer's email address. However, I don't necessarily think that it should be impossible for the customer themselves to change their email address. So instead of enforcing this validation at the service level my immediate thinking would be to put it in the API controller. 

Would be good to hear your thoughts?

Reply:
It shouldn’t the new changes should reflect that